app/src/main/java/com/github/se/gomeet/ui/mainscreens/profile/AddFriend.kt

```diff
@@ -0,0 +1,218 @@
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material.TopAppBar
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.automirrored.filled.ArrowBack
+import androidx.compose.material3.*
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalConfiguration
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import coil.compose.rememberAsyncImagePainter
+import com.github.se.gomeet.R
+import com.github.se.gomeet.model.user.GoMeetUser
+import com.github.se.gomeet.ui.mainscreens.events.GoMeetSearchBar
+import com.github.se.gomeet.ui.navigation.NavigationActions
+import com.github.se.gomeet.ui.navigation.Route
+import com.github.se.gomeet.viewmodel.UserViewModel
+import com.google.firebase.auth.ktx.auth
+import com.google.firebase.firestore.FirebaseFirestore
+import com.google.firebase.ktx.Firebase
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.tasks.await
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun AddFriend(nav: NavigationActions, userViewModel: UserViewModel) {
+
+  val screenWidth = LocalConfiguration.current.screenWidthDp.dp
+  val screenHeight = LocalConfiguration.current.screenHeightDp.dp
+  val query = remember { mutableStateOf("") }
+  var isLoading by remember { mutableStateOf(true) }
+  var users by remember { mutableStateOf(listOf<GoMeetUser>()) }
+  var followedUsers by remember { mutableStateOf(setOf<String>()) }
+
+  val coroutineScope = rememberCoroutineScope()
+
+  LaunchedEffect(Unit) {
+    coroutineScope.launch {
+      val fetchedUsers = userViewModel.getAllUsers()
+      val currentUser = Firebase.auth.currentUser!!.uid
+      if (fetchedUsers != null) {
+        users = fetchedUsers.filter { it.uid != currentUser } // Exclude the current user
+        // Initialize the followed users
+        val currentUserData = userViewModel.getUser(currentUser)
+        if (currentUserData != null) {
+          followedUsers = currentUserData.following.toSet()
+        }
+      }
+      isLoading = false
+    }
+  }
+
+  // Filtered users based on search query
+  val filteredUsers by
+      remember(query.value, users) {
+        derivedStateOf {
+          if (query.value.isEmpty()) {
+            users
+          } else {
+            users.filter { it.username.contains(query.value, ignoreCase = true) }
+          }
+        }
+      }
+
+  Scaffold(
+      topBar = {
+        Column {
```

Comment:
This Column is useless

Reply:
Removing it makes the Screen title and go back button overlap.